fix: LEAP-649: Saving draft on task switching

Gondragos opened the pull request:
This commit ensures the task ID remains consistent throughout the `onSubmitDraft` process, even while waiting for asynchronous operations.  Previously, the task ID could change during submission, leading to potential issues. This change prevents that.

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

Gondragos committed:
fix: saving draft on task switching

Preserve the task ID at the start of onSubmitDraft to ensure it's consistent while awaiting async operations. This prevents potential issues caused by task ID updates mid-submission.

Gondragos commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13928926918)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-LEAP-649/wrong-task-draft'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13928926918

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13928926918

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13947368297)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-LEAP-649/wrong-task-draft'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13947368297

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13947368297

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13964703376)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-LEAP-649/wrong-task-draft'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13964703376

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13964703376
